server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -387,9 +415,18 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
      */
     public void incomingRequest(final HttpRequest httpRequest, final HttpChannel httpChannel) {
         httpClientStatsTracker.updateClientStats(httpRequest, httpChannel);
+        final RequestTrackingHttpChannel trackingChannel = httpChannels.get(httpChannel);
         final long startTime = threadPool.rawRelativeTimeInMillis();
         try {
-            handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
+            // The channel may not be present if the close listener (set in serverAcceptedChannel) runs before this method because the
+            // connection closed early
+            if (trackingChannel == null) {
+                logger.warn("http channel [{}] missing tracking channel", httpChannel);
+                handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
```

Comment:
If the tracking channel has been removed, then doesn't that mean the httpChannel has been closed? How can the request actually be handled then?

Reply:
I switched that back to track down some persistent CI errors that I cannot reproduce locally.

Reply:
The CI errors were not related to this PR, I've switched this back but kept the warn.